packages/dd-trace/src/encode/0.4.js

```diff
@@ -8,8 +8,18 @@ const coalesce = require('koalas')
 
 const SOFT_LIMIT = 8 * 1024 * 1024 // 8MB
 
-function formatSpan (span) {
-  return normalizeSpan(truncateSpan(span, false))
+function formatSpan (span, encoder) {
+  span = normalizeSpan(truncateSpan(span, false))
+  // ensure span events are encoded as tags if agent doesn't support top level encoding
```

Comment:
Why not just pass the config at that point? It would be one less level of indirection.

Reply:
makes sense, fixing